tests/integration/common/data/db/test_constraints.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+from sqlalchemy.exc import IntegrityError
+
+from app.common.data.models import RoleEnum
+
+
+class TestUserRoleConstraints:
```

Comment:
Nice tests, good to have these checks on constraints when we're relying on them for such important things 👍